src/full/Agda/Syntax/Parser/Helpers.hs

```diff
@@ -148,11 +162,21 @@ mkName (i, s) = do
         alternating (_    :| x   : xs) = alternating $ x :| xs
         alternating (_    :|       []) = True
 
+-- | Create a name from a string
+mkName :: (Interval, String) -> Parser Name
+mkName = mkName' False
+
 -- | Create a qualified name from a list of strings
 mkQName :: [(Interval, String)] -> Parser QName
 mkQName ss = do
-    xs <- mapM mkName ss
-    return $ foldr Qual (QName $ last xs) (init xs)
+    let
+      ss0 = init ss
+      ss1 = last ss
```

Comment:
Preferable is `Agda.Utils.List1.initLast`: https://github.com/agda/agda/blob/ed9e11d1401d6006dd0693c3a51360e63ce1823d/src/full/Agda/Utils/List.hs#L130-L140